### DISCUSSION TITLE
Emulate 'no foreign keys allowed' for local development

### DISCUSSION THREAD

**Discussion by sandervankasteel** [planetscale/discussion#440] [Q&A] [answered]
Hi,

I am currently in the process of moving a Laravel application to PlanetScale and I was wondering if it there is a way to emulate the `SQLSTATE[HY000]: General error: 1105 foreign key constraints are not allowed, see https://vitess.io/blog/2021-06-15-online-ddl-why-no-fk/` error message for local development. 

I already tried to setting `SET GLOBAL foreign_key_checks=0;` in my local MySQL server, but alas, my local MySQL server doesn't throw an error when I try to create a table that has foreign key constraints.

**Comment by janpio** [comment, marked as answer]
Best way to achieve that is to use Vitess, which PlanetScale is running, instead of MySQL directly.
See this docker-compose example we are using at Prisma as an example: https://github.com/prisma/prisma/blob/fcc6d008fdc08cfdc5b7a1844df3950a516ab893/docker/docker-compose.yml#L33-L53

**Reply by sandervankasteel** [reply]
Thank you! I'll check it out :blush:.
